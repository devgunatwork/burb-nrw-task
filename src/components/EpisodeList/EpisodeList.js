import React from 'react';
import PropTypes from 'prop-types';
import List from '@material-ui/core/List';
import EpisodeItem from '../EpisodeItem';
import EPISODE_SHAPE from '../../shapes/episode.shape';

const EpisodeList = ({ episodes }) => {
  return (
    <List dense>
      {episodes.map((episode) => (
        <EpisodeItem episode={episode} key={episode.id} />
      ))}
    </List>
  );
};

EpisodeList.propTypes = {
  episodes: PropTypes.arrayOf(EPISODE_SHAPE).isRequired,
};

export default EpisodeList;
