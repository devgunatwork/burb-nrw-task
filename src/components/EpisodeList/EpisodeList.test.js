import React from 'react';
import EpisodeList from './EpisodeList';

describe('components', () => {
  describe('EpisodeList', () => {
    describe('UI', () => {
      it('should match the snapshot', () => {
        const wrapper = shallow(
          <EpisodeList
            episodes={[
              {
                id: 123,
                url: 'test url',
                name: 'name',
                season: 1,
                number: 1,
              },
            ]}
          />
        );

        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
