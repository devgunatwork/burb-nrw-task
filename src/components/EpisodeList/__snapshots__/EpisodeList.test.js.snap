// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components EpisodeList UI should match the snapshot 1`] = `
<WithStyles(ForwardRef(List))
  dense={true}
>
  <EpisodeItem
    episode={
      Object {
        "id": 123,
        "name": "name",
        "number": 1,
        "season": 1,
        "url": "test url",
      }
    }
    key="123"
  />
</WithStyles(ForwardRef(List))>
`;
